tasks/configure.yml

```diff
@@ -23,25 +26,33 @@
     setype: redis_conf_t
   when: redis_config_directory != "/etc"
 
-- name: "Set service name {{ item.redis_instance_name }}"
-  set_fact:
-    service_name: "{{ item.redis_instance_name }}.service"
-
-- name: "Set redis config for {{ item.redis_instance_name }}"
+- name: "Set redis config"
   template:
     src: redis.conf.j2
     dest: "{{ redis_config_directory }}/{{ item.redis_instance_name }}.conf"
     owner: "{{ redis_user_name }}"
     group: "{{ redis_group_name }}"
     mode: '0660'
     setype: redis_conf_t
-  notify: Restart redis
+  register: redis_config
+  loop: "{{ redis_configurations | default([]) }}"
+  loop_control:
+    label: "{{ item.redis_instance_name }}"
+
```

Comment:
What will happen if, for example, the first configuration file changed but the rest of the files or the last one did not? The registered **redis_config** ,as I know, will be overwritten in every loop, so the last **redis_config.changed** will be set to false, in this scenario, so the restart of the related service in the next task will not take place.

So the solution would be, in my opinion, to define the redis service name as a variable on the **Set redis config** task and notify it like this:

 ```yaml
register: redis_config
vars:
  service_name: "{{ item.redis_instance_name }}.service"
notify: Restart redis
loop: "{{ redis_configurations | default([]) }}"
loop_control:
   label: "{{ item.redis_instance_name }}"
 ```
 The handler **Restart redis** should then not be deleted.

Reply:
If at least one of the items in the loop changed, the (global) changed status will be true, see [Registering variables with a loop](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#registering-variables-with-a-loop) for details.

Reply:
Ah ok then my informations are wrong, all redis services will be restarted then at once, what should be ok for redis.